maintenance/projects/js-toolkit/packages/liferay-npm-build-tools-common/src/project/index.ts

```diff
@@ -384,36 +397,56 @@ export class Project {
 
 			return autoPresets;
 		}, []);
-	}
 
-	_loadDefaultPreset(): string {
-		let presetFilePath;
-
-		const autoPresets = this._findAutopresets();
-
-		if (autoPresets.length > 1) {
+		if (autopresets.length > 1) {
 			throw new Error(
 				'Multiple autopreset dependencies found in project (' +
-					autoPresets +
+					autopresets +
 					'): please remove the invalid ones or ' +
 					'explicitly define the preset to be used in the ' +
 					'.npmbundlerrc file'
 			);
 		}
-		else if (autoPresets.length) {
-			presetFilePath = this.resolve(autoPresets[0]);
+
+		return autopresets.length ? autopresets[0] : undefined;
+	}
+
+	_loadNpmbundlerrc(): void {
+		const npmbundlerrcPath = this._configFile.asNative;
+
+		const config = fs.existsSync(npmbundlerrcPath)
+			? readJsonSync(npmbundlerrcPath)
+			: {};
+
+		// Apply preset if necessary
+
+		let presetFilePath;
+
+		const autopreset = this._getAutopreset();
+
+		if (config.preset === undefined && autopreset) {
```

Comment:
Well, the check `typeof x === 'undefined'` was used because in early ES versions you could assign a value to `undefined` and change it during the execution of the program as many times as you wanted, making checks like `x === undefined` fail. But that was fixed long ago, so there's no need to do it any more, AFAIK.

And because `x === undefined` is like `x === null` or `x === 1`, we use that for coherence.

Also, I think the linters, TypeScript, etc. would complain if you assigned anything to `undefined`, so I think it's safe to use this construct.